src/plugins/formulas/engine/register.js

```diff
@@ -0,0 +1,186 @@
+import staticRegister from '../../../utils/staticRegister';
+import { isUndefined } from '../../../helpers/mixed';
+import { warn } from '../../../helpers/console';
+import { PLUGIN_KEY } from '../formulas';
+import { mergeEngineSettings } from './settings';
+
+/**
+ * Setups the engine instance. It either creates a new (possibly shared) engine instance, or attaches
+ * the plugin to an already-existing instance.
+ *
+ * @param {object} hotSettings Object containing the Handsontable settings.
+ * @param {string} hotId Handsontable guid.
+ * @returns {null|object} Returns the engine instance if everything worked right and `null` otherwise.
+ */
+export function setupEngine(hotSettings, hotId) {
```

Comment:
We only use `hotSettings` to retrieve the formulas plugin's settings, perhaps we could update the signature so that it would only consume `hotSettings[PLUGIN_KEY]`, instead of the whole hot settings object? Same with `mergeEngineSettings` and probably a few other functions.

Reply:
It's used in `mergeEngineSettings(hotSettings)`, which passes it onto `getEngineSettingsOverrides(hotSettings)`, which retrieves the `maxRows`, `maxCols` and `language` settings from the instance-level config.